### PR TITLE
Isolate indexes

### PR DESCRIPTION
**Pull request by dcousens**

### CODE REVIEW

File: example/service.js [hunk 1]
```diff
@@ -1,33 +1,19 @@
-let debug = require('debug')('example:service')
+let debug = require('debug')('service')
 let debugZmq = require('debug')('service:zmq')
 let debugZmqTx = require('debug')('service:zmq:tx')
-// let indexd = require('indexd') // XXX: you should use npm published package typically
-let indexd = require('../')
+let Indexd = require('../indexd') // FIXME: use require('indexd') in production
```
Comment (KryptoNova) [inline example/service.js]: This line change breaks the example.
I think it should be:
`let Indexd = require('../')`

**Comment by KryptoNova** [comment]
This is definitely an exciting PR.  Great work!!

I was testing this modification and I think the example service is broken.  I made a note on one problem that I found.  I think there is still another problem in the example.  I got an error in the socket.js file inside of the example.  See below:
```
socket.js:31
adapter.emitter.on('block', (blockId) => {
        ^

TypeError: Cannot read property 'emitter' of undefined
```

**Comment by dcousens** [comment]
@KryptoNova indeed thanks for testing!  The PR is still WIP,  but it is nearly done.
I'll ensure to ping you in when this is ready for merge (and therefore,  non-alpha-like testing too!).

**Comment by dcousens** [comment]
@KryptoNova go! This is ready for testing now.  I am using it for [bitcoinjs' regtest server for integration testing](https://github.com/bitcoinjs/private-bitcoin),  which is where I have moved the `example/` folder to.

**Pull request merged**